src/pyEQL/solution.py

```diff
@@ -37,6 +37,87 @@
 UNKNOWN_OXI_STATE = "unk"
 K_W = 1e-14  # ion product of water at 25 degC
 
+from pyEQL import ureg
+
+# Extend the UnitRegistry with custom units
+ureg.define('ppb = 1e-9 * mole / mole = parts_per_billion')
+ureg.define('ppm = 1e-6 * mole / mole = parts_per_million')
+ureg.define('% = 0.01 * mole / mole = percent')
```

Comment:
Good job figuring out how to hack the `UnitRegistry`! In principle this approach of using the unit registry could work, but in `get_amount` I've opted to manually translate the units into something `pint` understands, and that's the approach we should stick with in `preprocess_units`, so this block can be removed.